toolkit/smart-contracts/offchain/src/governance.rs

```diff
@@ -252,6 +252,41 @@ pub async fn get_governance_policy_summary<T: QueryLedgerState + QueryNetwork>(
 	}
 }
 
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct MultiSigParameters {
+	governance_authorties: Vec<MainchainKeyHash>,
+	threshold: u8,
+}
+
+impl MultiSigParameters {
+	pub fn new(governance_authorties: &[MainchainKeyHash], threshold: u8) -> Result<Self, &str> {
+		if governance_authorties.is_empty() {
+			return Err("governance authorities cannot be be empty");
+		}
+		if threshold == 0 {
+			return Err("threshold has to be a positive number");
+		}
+		if threshold as usize > governance_authorties.len() {
+			return Err("threshold cannot be greater than governance authorities length");
+		}
+		Ok(Self {
+			governance_authorties: governance_authorties.into_iter().map(|k| k.clone()).collect(),
+			threshold,
+		})
+	}
+
+	pub fn new_one_of_one(goveranance_authority: &MainchainKeyHash) -> Self {
+		Self { governance_authorties: vec![goveranance_authority.clone()], threshold: 1 }
+	}
+
+	pub(crate) fn to_simple_at_least_n(&self) -> SimpleAtLeastN {
```

Comment:
here maybe `as_*` will be better, due we do not consume self

Reply:
Done